extensions/amp-analytics/0.1/test/test-amp-analytics.js

```diff
@@ -1203,6 +1203,148 @@ describe('amp-analytics', function() {
+      const config = getConfig();
+      config.triggers.conditional.enabled = '${undefinedParam}';
+      const analytics = getAnalyticsTag(config);
+
+      return waitForNoSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.have.not.been.called;
+      });
+    });
+
+    it('does not allow a request through if a request param is missing', () => {
+      const config = getConfig();
+      config.triggers.conditional.enabled = '${queryParam(undefinedParam)}';
+      const analytics = getAnalyticsTag(config);
+
+      const urlReplacements = urlReplacementsForDoc(analytics.element);
+      sandbox.stub(urlReplacements.getVariableSource(), 'get').returns(null);
+
+      return waitForNoSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.have.not.been.called;
+      });
+    });
+
+
+    it('allows a request based on a variable when enabled on tag level', () => {
+      const config = getConfig();
+      config.enabled = '${foo}';
+      config.vars.foo = 'bar';
+      const analytics = getAnalyticsTag(config);
+
+      return waitForSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.be.calledOnce;
+      });
+    });
+
+    it('allows a request based on url-replacements ' +
+    'when enabled on tag level', () => {
+      const config = getConfig();
+      config.enabled = '${pageViewId}';
+      const analytics = getAnalyticsTag(config);
+
+      const urlReplacements = urlReplacementsForDoc(analytics.element);
+      sandbox.stub(urlReplacements.getVariableSource(), 'get').returns(0);
+      return waitForSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.be.calledOnce;
+      });
+    });
+
+    it('does not allow a request through when enabled on tag level', () => {
+      const config = getConfig();
+      config.enabled = '';
+      const analytics = getAnalyticsTag(config);
+
+      return waitForNoSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.have.not.been.called;
+      });
+    });
+
+    it('does not allow a request through if a variable is missing ' +
+    'when enabled on tag level', () => {
+      const config = getConfig();
+      config.enabled = '${undefinedParam}';
+      const analytics = getAnalyticsTag(config);
+
+      return waitForNoSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.have.not.been.called;
+      });
+    });
+
+    it('does not allow a request through if a request param is missing ' +
+    'when enabled on tag level', () => {
+      const config = getConfig();
+      config.enabled = '${queryParam(undefinedParam)}';
+      const analytics = getAnalyticsTag(config);
+
+      const urlReplacements = urlReplacementsForDoc(analytics.element);
+      sandbox.stub(urlReplacements.getVariableSource(), 'get').returns(null);
+
+      return waitForNoSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.have.not.been.called;
+      });
+    });
+  });
+
```

Comment:
Can you add a test or two for the case where both tag level and trigger level properties are specified?